src/main/cpp/JCryptominisat.cpp

```diff
@@ -0,0 +1,237 @@
+  (JNIEnv * env, jobject this_object, jlong p, jint lit1, jint lit2) {
+    auto lits = to_literals_vector(lit1, lit2);
+    decode(p)->add_clause(lits);
+  }
+
+JNI_METHOD(void, cryptominisat_1add_1clause__JIII)
+  (JNIEnv * env, jobject this_object, jlong p, jint lit1, jint lit2, jint lit3) {
+    auto lits = to_literals_vector(lit1, lit2, lit3);
+    decode(p)->add_clause(lits);
+  }
+
+JNI_METHOD(void, cryptominisat_1add_1clause__J_3I)
+  (JNIEnv * env, jobject this_object, jlong p, jintArray literals) {
+    decode(p)->add_clause(to_literals_vector(env, literals));
+  }
+
+JNI_METHOD(jint, cryptominisat_1solve__J)
+  (JNIEnv * env, jobject this_object, jlong p) {
+    return correctReturnValue(decode(p)->solve());
+  }
+
+JNI_METHOD(jint, cryptominisat_1solve__JI)
+  (JNIEnv * env, jobject this_object, jlong p, jint lit) {
+    auto lits = to_literals_vector(lit);
+    return correctReturnValue(decode(p)->solve(&lits));
+  }
+
+JNI_METHOD(jint, cryptominisat_1solve__JII)
+  (JNIEnv * env, jobject this_object, jlong p, jint lit1, jint lit2) {
+    auto lits = to_literals_vector(lit1, lit2);
+    return correctReturnValue(decode(p)->solve(&lits));
+  }
+
+JNI_METHOD(jint, cryptominisat_1solve__JIII)
+  (JNIEnv * env, jobject this_object, jlong p, jint lit1, jint lit2, jint lit3) {
+    auto lits = to_literals_vector(lit1, lit2, lit3);
+    return correctReturnValue(decode(p)->solve(&lits));
+  }
+
+JNI_METHOD(jint, cryptominisat_1solve__J_3I)
+  (JNIEnv * env, jobject this_object, jlong p, jintArray assumptions) {
+    auto lits = to_literals_vector(env, assumptions);
+    return correctReturnValue(decode(p)->solve(&lits));
+  }
+
+
+JNI_METHOD(jint, cryptominisat_1simplify__J)
+  (JNIEnv * env, jobject this_object, jlong p) {
+    return correctReturnValue(decode(p)->simplify());
+  }
+
+JNI_METHOD(jint, cryptominisat_1simplify__JI)
+  (JNIEnv * env, jobject this_object, jlong p, jint lit) {
+    auto lits = to_literals_vector(lit);
+    return correctReturnValue(decode(p)->simplify(&lits));
+  }
+
+JNI_METHOD(jint, cryptominisat_1simplify__JII)
+  (JNIEnv * env, jobject this_object, jlong p, jint lit1, jint lit2) {
+    auto lits = to_literals_vector(lit1, lit2);
+    return correctReturnValue(decode(p)->simplify(&lits));
+  }
+
+JNI_METHOD(jint, cryptominisat_1simplify__JIII)
+  (JNIEnv * env, jobject this_object, jlong p, jint lit1, jint lit2, jint lit3) {
+    auto lits = to_literals_vector(lit1, lit2, lit3);
+    return correctReturnValue(decode(p)->simplify(&lits));
+  }
+
+JNI_METHOD(jint, cryptominisat_1simplify__J_3I)
+  (JNIEnv * env, jobject this_object, jlong p, jintArray assumptions) {
+    auto lits = to_literals_vector(env, assumptions);
+    return correctReturnValue(decode(p)->simplify(&lits));
+  }
+
+JNI_METHOD(jboolean, cryptominisat_1get_1value)
+  (JNIEnv * env, jobject this_object, jlong p, jint lit){
+    return to_jboolean(decode(p)->get_model()[lit - 1]) ? lit : -lit;
```

Comment:
Isn't this function should return `jboolean`, but expression is of type `jint`? Seems like this compiles successfully only because of `int->bool` type coercion.